test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/PagedBufferedTextWriterTest.cs

```diff
@@ -265,6 +265,66 @@ public void Write_CharArray_Bounded_Null()
             Assert.Equal(3, pool.Returned.Count);
         }
 
+        [Fact]
+        public async Task FlushAsync_FlushesContent()
+        {
+            // Arrange
+            var pool = new TestArrayPool();
+            var inner = new StringWriter();
+
+            var writer = new PagedBufferedTextWriter(pool, inner);
+            for (var i = 0; i < Content.Length; i++)
+            {
+                writer.Write(Content[i]);
+            }
+
+            // Act
+            await writer.FlushAsync();
+
+            // Assert
+            Assert.Equal<char>(Content, inner.ToString().ToCharArray());
+        }
+
+        [Fact]
+        public async Task FlushAsync_WritesContentToInner()
```

Comment:
Could add the new assertion to the previous test. But they're really looking for unrelated things in the same scenario. Could change the previous test's name and merge the two. LMK if you feel strongly...